Bottom: ba8d27015d59d715a442fea45cf641120f2804d2
Top:    8f3456077c3be8e94984b5c19fa6bfb923d5eb11
Author: Ben Greear <greearb@candelatech.com>
Date:   2013-01-29 22:48:25 -0800

nfs: Pass srcaddr into mount request.

This starts tying the srcaddr logic together.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/fs/nfs/internal.h b/fs/nfs/internal.h
index 2c585c3..8a6de8a 100644
--- a/fs/nfs/internal.h
+++ b/fs/nfs/internal.h
@@ -127,6 +127,7 @@ struct nfs_parsed_mount_data {
 /* mount_clnt.c */
 struct nfs_mount_request {
 	struct sockaddr		*sap;
+	struct sockaddr         *srcaddr;
 	size_t			salen;
 	char			*hostname;
 	char			*dirpath;
diff --git a/fs/nfs/mount_clnt.c b/fs/nfs/mount_clnt.c
index 09b1900..662347f 100644
--- a/fs/nfs/mount_clnt.c
+++ b/fs/nfs/mount_clnt.c
@@ -156,6 +156,7 @@ int nfs_mount(struct nfs_mount_request *info)
 	struct rpc_create_args args = {
 		.net		= info->net,
 		.protocol	= info->protocol,
+		.saddress	= info->srcaddr,
 		.address	= info->sap,
 		.addrsize	= info->salen,
 		.servername	= info->hostname,
diff --git a/fs/nfs/super.c b/fs/nfs/super.c
index 1abca0e..021c1dc 100644
--- a/fs/nfs/super.c
+++ b/fs/nfs/super.c
@@ -1725,6 +1725,8 @@ static int nfs_request_mount(struct nfs_parsed_mount_data *args,
 	struct nfs_mount_request request = {
 		.sap		= (struct sockaddr *)
 						&args->mount_server.address,
+		.salen		= args->mount_server.addrlen,
+		.srcaddr	= (struct sockaddr *)&args->srcaddr.address,
 		.dirpath	= args->nfs_server.export_path,
 		.protocol	= args->mount_server.protocol,
 		.fh		= root_fh,
